src/chainrules.jl

```diff
@@ -286,5 +286,48 @@ function ChainRulesCore.rrule(::typeof(pd_from_upper), X::AbstractMatrix)
     end
 end
 
-# Fixes Zygote's issues with `@debug`
+# Fixes AD issues with `@debug`
 ChainRulesCore.@non_differentiable _debug(::Any)
+
+# ChainRules for utility functions used by PlanarLayer
```

Comment:
Are these copied over from somewhere else, or new code? Do we need them? I think `ChainRules` rules are mostly for Zygote, although other AD packages use them too, so wondering if/why these are necessary.

Reply:
Hmm, odd. Happy to have a look at the error and figure it out together if it helps.

Reply:
Absolutely. Ping me or Penny on Slack when you have a moment and let's have a look at it together.